homeassistant/components/lg_netcast/media_player.py

```diff
@@ -54,23 +60,45 @@
 )
 
 
-def setup_platform(
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a LG Netcast Media Player from a config_entry."""
+
+    host = config_entry.data[CONF_HOST]
+    access_token = config_entry.data[CONF_ACCESS_TOKEN]
+    unique_id = config_entry.unique_id
+    name = config_entry.data.get(CONF_NAME, DEFAULT_NAME)
+    model = config_entry.data[CONF_MODEL]
+
+    client = LgNetCastClient(host, access_token)
+
+    hass.data[DOMAIN][config_entry.entry_id] = client
+
+    async_add_entities([LgTVDevice(client, name, model, unique_id=unique_id)])
+
+
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the LG TV platform."""
 
     host = config.get(CONF_HOST)
-    access_token = config.get(CONF_ACCESS_TOKEN)
-    name = config[CONF_NAME]
-    on_action = config.get(CONF_ON_ACTION)
 
-    client = LgNetCastClient(host, access_token)
-    on_action_script = Script(hass, on_action, name, DOMAIN) if on_action else None
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+    )
 
-    add_entities([LgTVDevice(client, name, on_action_script)], True)
+    if (
+        result.get("type") == FlowResultType.ABORT
+        and result.get("reason") == "cannot_connect"
+    ):
+        raise PlatformNotReady(f"Connection error while connecting to {host}")
```

Comment:
I think we should also raise an issue here, telling the user the import is pending because we can not connect to the TV and will be retried, the issue should be removed when import suceeds.

Reply:
This is handled in the config_flow.py

Reply:
OK, in that case I don't think we need to check the return value of the import flow here.

Reply:
Whilst the issue is now raised in the config_flow.py, the config_flow.py doesn't handle the waiting for the TV to be turned on and connected to the network to finish configuring the device. (This is because as part of the migration we look-up the UUID of the TV that's otherwise not present in the YAML config). Without the UUID the migration cannot proceed.

I'm using the PlatformNotReady Exception to trigger the  
Removing this check will require the user to restart Home Assistant with the TV online for the migration to occur. 

All other Aborts indicate an invalid or already migrated YAML configuration.

If this behaviour is undesirable or needs to be implemented in another way - let me know and I will adjust appropiately.

Reply:
Sure, that explanation makes perfect sense 👍